src/ConstructiveSolidGeometry/VolumePrimitives/HexagonalPrism.jl

```diff
@@ -0,0 +1,86 @@
+struct HexagonalPrism{T,TR,TZ} <: AbstractVolumePrimitive{T}
+    r::TR
+    z::TZ
+    
+    function HexagonalPrism( ::Type{T},
+                             r::Union{T, <:AbstractInterval{T}},
+                             z::Union{T, <:AbstractInterval{T}}) where {T}
+        new{T,typeof(r),typeof(z)}(r,z)
+    end
+end
+
+# Constructors
+function HexagonalPrism(; rInner = 0, rOuter = 1, zMin = -1, zMax = 1)
+    T = promote_type(typeof.((rInner, rOuter, zMin, zMax))...)
+    r = rInner == 0 ? T(rOuter) : T(rInner)..T(rOuter)
+    z = zMax == -zMin ? T(zMax) : T(zMin)..T(zMax)
+    HexagonalPrism( T, r, z)
+end
+HexagonalPrism(rInner, rOuter, zMin, zMax) = HexagonalPrism(; rInner, rOuter, zMin, zMax)
+
+function HexagonalPrism(rOuter::R, height::H) where {R<:Real, H<:Real}
+    T = promote_type(R,H)
+    HexagonalPrism( T, T(rOuter), T(height/2))
+end
+
+in(p::CartesianPoint, hp::HexagonalPrism{<:Any, <:Real, <:Any}) =
+    _in_z(p, hp.z) && _in_x(p, hp.r * sqrt(3)/2) && _in_y(p, hp.r - abs(p.x) /sqrt(3))
```

Comment:
`sqrt(T(3))/T(2)` is probably better. I am not sure if the compiler is clever enough. 
Did you testes what happens if `T = Float32`. Plain `sqrt(3)` should be Float64.

Reply:
`sqrt(T(3))/2` is already enough

Reply:
Basically the same as above. Test it by putting it into a function (not just the REPL) and look at the produced code via `@code_llvm` and `@code_native` besides `@code_warntype`.